Details on Helm 3 mentioned (#74)

jodavis-suse opened the pull request:
Include notes for Helm 3, expected Helm 2 end of support, and
a warning about migration.

jodavis-suse committed:
Details on Helm 3 mentioned (#74)

Include notes for Helm 3, expected Helm 2 end of support, and
a warning about migration.

jodavis-suse commented on adoc/release-4.5.0.adoc:
I think this is sufficient, as moving in and out of the 'internal' network is something the user will do anyway to pick up updated containers and packages.

jodavis-suse committed:
Add detail on Helm 3 enhancements (#74)

kkaempf commented:
:+1: Thanks !

suseclee commented on adoc/release-4.5.0.adoc:
```diff
@@ -45,13 +45,37 @@ After this you will be able to perform all the `skuba` commands that you would c
+* One important change is the removal of Tiller, which simplifies deployment and management.
+* Configuration is now stored in Kubernetes Secrets rather than ConfigMaps.
+* Chart releases are now scoped to Kubernetes namespaces.
+* Charts can now be installed from link:https://hub.helm.sh/[the Helm Hub site] and from user configured repositories.
```
https://hub.helm.sh provides information of the charts.
So someone might be misunderstanding from this. 
For example, if I go to https://hub.helm.sh/charts/stable/aerospike, it shows how to install the aerospike like  "helm install stable/aerospike --version 0.3.2" And User has to add the stable repo because helm3 does not include stable repo as a default.

suseclee replied:
I think " Charts can be searched for in Helm Hub or repositories using the `helm search` command." should be enough.

jodavis-suse replied:
Thanks. I removed the line. It felt like the list was a little long anyway, so simpler is better. :)

suseclee commented on adoc/release-4.5.0.adoc:
```diff
@@ -45,13 +45,37 @@ After this you will be able to perform all the `skuba` commands that you would c
+
+These changes and more are documented by the community in the link:https://helm.sh/docs/faq/#changes-since-helm-2[Helm FAQs] and link:https://v3.helm.sh/docs/topics/v2_v3_migration/[Migrating 2 to 3 document].
+
+Helm 2 is still installed by default to avoid forcing a migration of chart releases when upgrading {productname}.
```
Good to mention in release note !!

jodavis-suse replied:
Yes, I'll give Klaus credit for requesting that. :)

suseclee commented on adoc/release-4.5.0.adoc:
```diff
@@ -45,13 +45,37 @@ After this you will be able to perform all the `skuba` commands that you would c
+Helm 2 is still installed by default to avoid forcing a migration of chart releases when upgrading {productname}.
+The Helm 3 version is not a drop in upgrade of Helm 2, though most charts written for "apiVersion: v1" will work with Helm 3.
+However, as the stated end of community support for Helm 2 is November 13, 2020, *you should migrate to Helm 3 as soon as reasonable*.
+We recommend for fresh installations using Helm 3 from the start.  See link:https://helm.sh/blog/covid-19-extending-helm-v2-bug-fixes/[Official Helm blog].
```
Good to have this recommendation.

suseclee reviewed (commented):
Except "* Charts can now be installed from link:https://hub.helm.sh/[the Helm Hub site] and from user configured repositories." line, everything looks great. Thanks for making it in a short time.

jodavis-suse committed:
Simplify list of Helm 3 features (#74)

The "installed from" line could be a little misleading, and is
somewhat redundant with the next line about search.

suseclee reviewed (approved):
LGTM
